tests/unit/data/data_pipeline/test_sample.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+import torch
+
+from fairseq2.data import DataPipeline, read_sequence
+from fairseq2.data.text.text_reader import read_text
+from fairseq2.utils.version import is_pt2_or_greater
+
+
+@pytest.mark.skipif(
+    not is_pt2_or_greater(),
+    reason="different sampling results with versions lower than PyTorch 2.0.0",
+)
+class TestSampleOp:
```

Comment:
@cbalioglu what's up with test classes ?
pytest doesn't need those. And with one file per op like we do, filename + function name is enough to identify the test case.

(less indent == nicer black formatting)

Reply:
Old habits die hard, so used to having classes in `unittest`, and `googletest`, just adding them without giving thought. Feel free to remove them. I don't really have any strong preference.

Reply:
I am closing this since it's not related particularly to this PR.